stdpopsim/models.py

```diff
@@ -137,6 +137,176 @@ def asdict(self):
             "migration_matrix": self.migration_matrix,
             "demographic_events": self.demographic_events}
 
+# ###############################################################
+# UNDER CONSTRUCTION - GROUND TRUTH N_t
+# ###############################################################
+
+    # TODO Rename
+    def get_Ne_through_time_single_pop(self, end, num_steps=10):
+        """
+        This function computes the defined Ne for each individual
+        sub population for any demographic model.
+
+        `end` specifies at which generation to stop
+        collecting effective population sizes among subpopulations
+
+        `num_steps` parameter will determine how many points between
+        `start` and `end` to sample.
+
+        Returns a tuple where the first value
+        is the steps (in generations) and the second is a numpy
+        ndarray that will contain the effective population sizes for
+        each subpopulation individually, for each step, as defined
+        by the model which inherits from this class.
+        """
+
+        num_pops = self.num_pops()
+        N_t = np.zeros([num_steps, num_pops])
+        steps = np.linspace(0, end, num_steps)
+        for j, t in enumerate(steps):
+            N = self._get_N_M_at_t(t)["population_sizes"]
+            N_t[j] = N
+        return steps, N_t
+
+    # TODO Rename
+    def get_Ne_through_time_all_pops(self, end, num_samples_per_location, num_steps=10):
+        """
+        This function will calculate the ground truth
+        effective population size (Ne) for a population with
+        multiple sub-populations. This takes a
+        demographic model and returns a
+        function of time that gives inverse coalescence rate.
+
+        `num_samples_per_location` should be a list containing the
+        number of samples in each subpopulation organized by population index.
```

Comment:
Vote to rename this `num_samples` with documentation: "`num_samples` should be a list of the same length as the number of populations, so that `num_samples[j]` is the number of samples in subpopulation `j`."